File: janus/src/main/java/com/serenegiant/janus/response/videoroom/ListResponse.java
package com.serenegiant.janus.response.videoroom;

import java.util.Arrays;

import androidx.annotation.NonNull;

/**
 * listリクエストの結果
 */
public class ListResponse<T> {
	public final String videoroom;
	public final T[] list;

	public ListResponse(final String videoroom, final T[] list) {
		this.videoroom = videoroom;
		this.list = list;
	}

	@NonNull
	@Override
	public String toString() {
		return "ListResponse{" +
			"videoroom='" + videoroom + '\'' +
			", list=" + Arrays.toString(list) +
			'}';
	}
}

File: janus/src/main/java/com/serenegiant/janus/response/videoroom/ListResponse.kt
package com.serenegiant.janus.response.videoroom
/*
 * JanusRTCAndroid
 * Video chat sample app using videoroom plugin on janus-gateway server and WebRTC.
 *
 * Copyright (c) 2018 - 2022 saki t_saki@serenegiant.com
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

/**
 * listリクエストの結果
 */
class ListResponse<T>(val videoroom: String, val list: Array<T>) {
	override fun toString(): String {
		return "ListResponse{" +
			"videoroom='" + videoroom + '\'' +
			", list=" + list.contentToString() +
			'}'
	}
}
